### PR TITLE
fix: rate limiting

### PR DESCRIPTION
**Pull request by bonhokage06**
# Remove wallet address limiting and use client address instead.

### Before Proceeding you should able to run eth-faucet (Read the link below)

[https://www.notion.so/lilypadnetwork/Faucet-Local-Development-6006fd62fdef4e3e90da745210fe1b1c](https://www.notion.so/lilypadnetwork/Faucet-Local-Development-6006fd62fdef4e3e90da745210fe1b1c)

### Requirements
1. Run dev chain in your localhost and add some funds in the testing wallets.
2.  Install cloudflared in your testing environment with this commands in linux

        curl -L --output cloudflared.deb https://github.com/cloudflare/cloudflared/releases/latest/download/cloudflared-linux-<your computer architecture>.deb
      
        sudo dpkg -i cloudflared.deb

### Instructions how to setup for localhost testing
1.  Modify the stack file and add this codes

        function faucet-docker-run-local() {
        docker rm -f faucet || true
        docker run \
          --name faucet \
          -p 8082:8080 \
          -e DOPPLER_TOKEN="$(doppler configs tokens create -p faucet -c <doppler_faucet_config> docker --max-age 1m --plain)" \
          faucet 
            }

3. Modify the Dockerfile and update expose-$expose_via with this code

          FROM expose-$expose_via AS final
          RUN echo "doppler run -- eth-faucet --faucet.name=dev --faucet.amount=1 --faucet.tokenamount=20 -- 
          faucet.minutes=1" 
          >> run.sh
          RUN chmod +x run.sh
           CMD ["/bin/bash", "./run.sh"]`

4. In cmd/server.go update ChainIDMap variable with this ( to add development chainId)

        chainIDMap = map[string]int{"goerli": 5, "sepolia": 11155111, "dev": 412346}

5. Build and run the faucet locally using this commands
  
        ./stack faucet-docker-build
        ./stack faucet-docker-run-local
       
5. In eth-faucet folder kindly run this command to be able to connect to Cloudflare tunnel
       
          cloudflared tunnel run --token eyJhIjoiZTUzY2IzMjkzMDE3YThjMThiYmU4ZDMzYWFlMDUwYWMiLCJ0IjoiM2RjYzdkMTktN2JiZC00NjFhLTg0Y2UtN2QzYWM3NzVlOTdiIiwicyI6IlpHRTRaV1ZqT1RjdE9EZGxZUzAwTkdaakxXSTNOemN0WXpRMk1XTmhNVE14Wm1ZMSJ9 --url http://localhost:8082
6.  Open this URL [https://test1.lilypadnetwork.xyz/](https://test1.lilypadnetwork.xyz)

# Testing
1. Open 2 browsers with one of them connected to Vpn
2. Request tokens for both until rate limited. You can use local wallet addresses.
3. It should have a different cooldown time. See image below:

result

### CODE REVIEW

File: internal/server/middleware.go [hunk 1]
```diff
@@ -51,11 +51,10 @@ func (l *Limiter) ServeHTTP(w http.ResponseWriter, r *http.Request, next http.Ha
 
 	clintIP := getClientIPFromRequest(l.proxyCount, r)
 	l.mutex.Lock()
-	if l.limitByKey(w, address) || l.limitByKey(w, clintIP) {
+	if l.limitByKey(w, clintIP) {
```
Comment (alvin-reyes) [inline internal/server/middleware.go:54]: clintIP or clientIP?

Comment (bonhokage06) [reply internal/server/middleware.go]: Hi sir @alvin-reyes  in the original Code it is 'clintIP'. I already updated it now into 'clientIP'. thank you

**Comment by kquirapas** [comment]
@bonhokage06 

Any changes to documentation once this gets merged?

**Comment by bonhokage06** [comment]
> @bonhokage06
> 
> Any changes to documentation once this gets merged?

Yes @kquirapas we need to update our "local faucet development" docs.

**Pull request merged**